larrecodnn/ImagePatternAlgs/NuSonic/Triton/TritonClient.cc

```diff
@@ -26,15 +26,36 @@ namespace lartriton {
     : allowedTries_(params.get<unsigned>("allowedTries", 0))
     , serverURL_(params.get<std::string>("serverURL"))
     , verbose_(params.get<bool>("verbose"))
+    , ssl_(params.get<bool>("ssl"))
     , options_(params.get<std::string>("modelName"))
   {
     //get appropriate server for this model
     if (verbose_) MF_LOG_INFO("TritonClient") << "Using server: " << serverURL_;
 
     //connect to the server
-    //TODO: add SSL options
-    triton_utils::throwIfError(nic::InferenceServerGrpcClient::Create(&client_, serverURL_, false),
-                               "TritonClient(): unable to create inference context");
+    if (ssl_) {
+      std::string root_certificates;
+      std::string private_key;
+      std::string certificate_chain;
+      nic::SslOptions ssl_options = nic::SslOptions();
+      ssl_options.root_certificates = root_certificates;
+      ssl_options.private_key = private_key;
+      ssl_options.certificate_chain = certificate_chain;
```

Comment:
It is necessary to enable ssl (set to true) when using the triton server on EAF.  Burt H. says root_certificates, private_key, and certificate_chain can be left empty strings since we are not doing any auth or client-side verification of the cert, we're only requesting to encrypt the datastream.  But I have modified the code to allow these strings to be set from fcl parameters if the user needs to do so.